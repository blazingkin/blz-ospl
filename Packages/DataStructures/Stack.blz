constructor Stack
    stack = []
    i = -1

    :push(item)
        stack[i++] = item
    end

    :pop
        i--
        stack[i + 1]
    end

    :length
        size()
    end

    :size
        i + 1
    end

    :show
        if i < 0
            [].show()
        else
            stack.slice(0, i + 1).show()
        end
    end

    :==(other)
        if this.i == other.i
            for k = 0; k < this.i; k++
                if this.stack[i] != other.stack[i]
                    return false
                end
            end
            return true
        end
        return false
    end

end