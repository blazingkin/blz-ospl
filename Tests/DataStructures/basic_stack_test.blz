import DataStructures

first = Stack()
second = Stack()

print(first == second) # Empty stacks should be equal

first.push(1)
first.push(2)
print(first.pop()) # should be 2
print(first.pop()) # should be 1

first.push(1)

print(first == second) # They are now not equal
second.push(1)
print(first == second) # Now equal
second.push("a")
first.push("a")
print(first == second) # Still equal