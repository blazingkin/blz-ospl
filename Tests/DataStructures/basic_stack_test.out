true
2
1
false
true
true
